src/pages/quiz-preview/QuizPreview.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState, useCallback } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useParams, useNavigate } from 'react-router-dom'
+
+import Box from '@mui/material/Box'
+import Divider from '@mui/material/Divider'
+
+import Loader from '~/components/loader/Loader'
+import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import { StartViewQuizInfo } from '~/containers/quiz/quiz-info/QuizInfo'
+import Button from '~scss-components/button/Button'
+import QuizInfoSection from '~/containers/quiz/quiz-info-section/QuizInfoSection'
+import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import TimeLimitReminder from '~/containers/quiz/time-limit-reminder/TimeLimitReminder'
+
+import useQuery from '~/hooks/use-query'
+
+import { ResourceService } from '~/services/resource-service'
+import styles from '~/pages/quiz-preview/QuizPreview.styles'
+import { defaultQuizResponse } from '~/pages/quiz/Quiz.constant'
+
+import { formatTime, getFormattedDate } from '~/utils/helper-functions'
+import { Typography } from '@mui/material'
+import useQuizQuery from '~/hooks/query/use-quiz-query'
+
+const QuizPreviewPage: React.FC = () => {
+  const { id: cooperationId = '', quizId = '' } = useParams()
+  const navigate = useNavigate()
+  const [isOpen, setIsOpen] = useState(false)
+
+  const { t } = useTranslation()
+
+  const { quiz, isLoading } = useQuizQuery(quizId)
+
+  const openModal = useCallback(() => {
+    setIsOpen(true)
+  }, [])
+
+  const {
+    settings: { attemptLimit, timeLimit },
+    description,
+    title,
+    items
+  } = quiz ?? defaultQuizResponse
+
+  const getQuizzes = useCallback(() => {
```

Comment:
A lot of inconsistently here. Query function is `getQuizzes`, but service named get**FinishedQuizzes**ByQuizId, variable for that named `finishedAttempts`, and query key is `finished-quizzes`. Please, name those variables consistently. It will be more readable and understandable.
```suggestion
  const getFinishedQuizzes = useCallback(() => {
```